### PR TITLE
Phantomcss visualtesting

### PR DESCRIPTION
**Pull request by joschect**
#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [ ] Include a change request file if publishing 
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

Add in Phantomcss visual testing

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by micahgodbolt** [comment]
@dzearing thinking of naming conventions for the Button test page.

I'm thinking either 
`Button.visualtest.Page.tsx` 
or 
`ButtonPage.visualtest.tsx`

I like the second one best as it'll mean Button.xxx refers to the component and ButtonPage.xxx refers to documentation.

File: ghdocs/VISUALTESTS.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+Visual testing is still in the early stages and is subject to change.
+
+Our tests are visual tests are built with with [PhantomCSS](https://github.com/Huddle/PhantomCSS) which makes extensive use of [Casperjs](http://casperjs.org/).
+
+In order to have PhantomCSS run you must have python version 2.6 or higher installed.
+
+To run tests:
+
+1. Navigate to git/office-ui-fabric-react/packages/office-ui-fabric-react in command prompt.
```
Comment (micahgodbolt) [inline ghdocs/VISUALTESTS.md]: This assumes where the repo is placed and what the folder is named.

File: ghdocs/VISUALTESTS.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+2. Run gulp visualtest --production
```
Comment (micahgodbolt) [inline ghdocs/VISUALTESTS.md]: run `gulp visualtest --production`

Comment (micahgodbolt) [inline ghdocs/VISUALTESTS.md]: or without the $, whatever is standard

File: packages/office-ui-fabric-react/src/VisualTestRoot.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+/* tslint:enable:no-unused-variable */
+import * as ReactDOM from 'react-dom';
+import { VisualTestState } from './VisualTestState';
+import { Route, Router } from './utilities/router/index';
+import { setBaseUrl } from '@uifabric/utilities/lib/resources';
+import './VisualTestRoot.scss';
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/VisualTestRoot.tsx]: i don't see this file checked in

File: packages/office-ui-fabric-react/src/components/Button/Button.visualtest.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { Casper, IPhantomCSS } from '../../visualtest/PhantomCssInterface';
+import { baseUrl } from '../../common/VisualTest'
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Button/Button.visualtest.ts]: missing `;`

File: packages/office-ui-fabric-react/src/components/Button/Button.visualtest.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    phantomcss.screenshot('#DefaultButton', 'Button_not_pressed');
+  }).then(function () {
+    this.mouse.move('#DefaultButton');
+    phantomcss.screenshot('#DefaultButton', 'Button_Hovered');
+  }).then(function () {
+    this.mouse.down('#DefaultButton');
+    phantomcss.screenshot('#DefaultButton', 'Button_pressed');
+  }).then(function () {
+    phantomcss.screenshot('#IconButton', 'Icon_Button');
+  })
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Button/Button.visualtest.ts]: missing `;`

File: packages/office-ui-fabric-react/gulpfile.js [hunk 6]
```diff
@@ -16,6 +19,27 @@ let {
...
+    port: 43210,
+    livereload: false,
+    directoryListing: false
+  });
+  if (!options.args['debug']) {
+    let casperJs = require('gulp-phantomcss');
+    gulp.src(['lib/**/*.visualtest.js'])
+      .pipe(casperJs(
+        {
+          screenshots: 'visualtests',
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/gulpfile.js]: "baselines" is a more common name for this file. Visualtests make me think the tests go here, not the baseline images.

**Comment by micahgodbolt** [comment]
Other than a few suggestions, it looks great!

File: ghdocs/VISUALTESTS.md (hunk 1 above)
Comment (micahgodbolt) [inline ghdocs/VISUALTESTS.md]: @joschect can you change this to just `packages/office-ui-fabric-react`. Rather not assume everyone is putting things into their git folder, or naming them the same.

**Comment by micahgodbolt** [comment]
@dzearing @mikewheaton do you want to put some eyes on this before we merge it in? Folder locations, naming conventions...

I'm bout ready to hit the button once we get a passing build.

**Comment by joschect** [comment]
Alrighty then! Imma complete dis!

**Pull request merged**